lib/client.js

```diff
@@ -135,17 +135,18 @@ Client.prototype.close = function(){
  * @param {Object} packet object
  * @param {Boolean} whether packet is already encoded
  * @param {Boolean} whether packet is volatile
+ * @param {Boolean} whether packet should be compressed
  * @api private
  */
 
-Client.prototype.packet = function(packet, preEncoded, volatile){
+Client.prototype.packet = function(packet, preEncoded, volatile, compress){
   var self = this;
 
   // this writes to the actual connection
   function writeToEngine(encodedPackets) {
     if (volatile && !self.conn.transport.writable) return;
     for (var i = 0; i < encodedPackets.length; i++) {
-      self.conn.write(encodedPackets[i]);
+      self.conn.write(encodedPackets[i], { compress: compress });
```

Comment:
Just curious, is this the same signature the WS client would use? I mention this since it'd be nice to maintain parity so that you could technically replace `engine.io-client` with a `WebSocket` client altogether.